model/minsky.cc

```diff
@@ -288,78 +288,87 @@ namespace minsky
     GroupPtr g(new Group);
     g->self=g;
     m.populateGroup(*g);
+    // stash values of parameters in copied group, as they are reset for some unknown reason later on. for ticket 1258
+    vector<pair<string,string>> existingParms; 
+    for (auto& i: g->items) 
+        if (i->variableCast() && i->variableCast()->type()==VariableType::parameter) existingParms.push_back(make_pair(i->variableCast()->valueId(),i->variableCast()->init()));
```

Comment:
i->variableCast() is not a trivial call - there is no need to call it 4 times.

Reply:
Use if (auto v=i->variableCast())
          if (v->type() ... v->valueId(), v->init())
pattern

Reply:
Because "auto v=i->variableCast()" does not seem to always work well, I'm trying "VariablePtr v=i->variableCast()". As far as I've read this should be fine, because we know that the type variableCast() returns has to be derived from VariableBase?

Reply:
Not why "auto v=" is a problem - we use it elsewhere just fine. But variableCast returns a VariableBase*, so that is what you should use if you go that route, not VariablePtr, which would completely muck up ownership.

Reply:
> Use if (auto v=i->variableCast())
> if (v->type() ... v->valueId(), v->init())
> pattern

It's the "if (auto ...)" that is the problem. If I instead do

auto v=i->variableCast()
if (v)
...

then it still works. I'll commit the updated code now.

Reply:
> Not sure why "auto v=" is a problem - we use it elsewhere just fine. But variableCast returns a VariableBase*, so that is what you should use if you go that route, not VariablePtr, which would completely muck up ownership.

Thank you for explaining that. I quickly ran into trouble (segfault) when I did that and tested copying and pasting in 1Free.mky.